modules/service/src/main/scala/lucuma/odb/service/GuideEnvironmentService.scala

```diff
@@ -129,13 +129,15 @@ object GuideEnvironmentService {
     constraints:        ConstraintSet,
     posAngleConstraint: PosAngleConstraint,
     optWavelength:      Option[Wavelength],
-    fpu:                Either[GmosNorthFpu, GmosSouthFpu],
+    optFpu:             Option[Either[GmosNorthFpu, GmosSouthFpu]],
     explicitBase:       Option[Coordinates]
   ) {
-    def agsParams: AgsParams                  = AgsParams.GmosAgsParams(fpu.some, PortDisposition.Side)
+    def agsParams: AgsParams                 = AgsParams.GmosAgsParams(optFpu, PortDisposition.Side)
```

Comment:
I guess we're anticipating a refactor when we introduce some other instrument?

Reply:
Yeah. I think we're going to need to do some refactoring to get whatever information is needed for other `AgsParams` types. Right now, there is only `GmosAgsParams`. We'll also need to have different guide probe(s), etc. If you know of things that would help mitigate that future refactoring, I could do those now. Maybe `guideProbe` could be a value in `ObservationInfo`? But, I don't really know how other instruments will work.

Reply:
Oh, now that I look, the `guideProbe` is part of `AgsAnalysis.Usable`. I missed that before. I can switch to that.

Reply:
Actually, I'll do that in my next PR as I'm making some changes that will make that easier.